File: src/application/dtos/package/lend-packages.dto.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { IsInt, IsNotEmpty, IsOptional, IsString, IsUUID, Min } from 'class-validator';

export class LendPackagesDto {
  @ApiProperty({ example: 'uuid-customer-id' })
  @IsNotEmpty({ message: "L'ID du client est requis" })
  @IsUUID('4', { message: "L'ID du client doit être un UUID valide" })
  customerId: string;

  @ApiProperty({ example: 3, minimum: 1, description: "Nombre d'emballages à prêter" })
  @IsNotEmpty({ message: "Le nombre d'emballages est requis" })
  @IsInt({ message: "Le nombre d'emballages doit être un entier" })
  @Min(1, { message: "Le nombre d'emballages doit être supérieur à 0" })
  quantity: number;

  @ApiPropertyOptional({ example: 'Prêt lors de la livraison du 25/04' })
  @IsOptional()
  @IsString()
  notes?: string;
}

File: fopa_sop_apk/lib/features/feat_package/datas/dtos/lend_packages_dto.dart
class LendPackagesDto {
  final String customerId;
  final int quantity;
  final String? notes;

  LendPackagesDto({
    required this.customerId,
    required this.quantity,
    this.notes,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{
      'customerId': customerId,
      'quantity': quantity,
    };
    if (notes != null) map['notes'] = notes;
    return map;
  }
}
